packages/loader/container-loader/src/container.ts

```diff
@@ -1529,6 +1535,10 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             }
         });
 
+        deltaManager.once("submitOp", (message: IDocumentMessage) => {
+            this.connectionStateHandler.clientSentOps(this._deltaManager.connectionMode);
```

Comment:
If this._deltaManager.connectionMode is not connected, then we will get nack. This op has no chance of making it through (i.e. we do not care about tracking it really).
Maybe API shape is fine, but then I'd make it noop in ConnectionStateHandler if it's not connected state.
Mostly because it will reduce possible states, we do not care about that state (thus less chances for bugs)